src/Access/EnabledQuota.cpp

```diff
@@ -52,22 +48,24 @@ struct EnabledQuota::Impl
             return end;
         }
 
+        bool initial_end = !end_loaded.count(); /// whether the end of the interval wasn't calculated before
         const auto duration = interval.duration;
 
         do
         {
             end = end + (current_time - end + duration) / duration * duration;
             if (end_of_interval.compare_exchange_strong(end_loaded, end.time_since_epoch()))
-            {
-                boost::range::fill(interval.used, 0);
                 break;
-            }
             end = std::chrono::system_clock::time_point{end_loaded};
         }
         while (current_time >= end);
 
-        if (counters_were_reset)
-            *counters_were_reset = true;
+        if (!initial_end)
+        {
+            boost::range::fill(interval.used, 0);
+            if (counters_were_reset)
```

Comment:
Could you please add some comments about `counters_were_reset` variable ?

Reply:
done